### PR TITLE
Only post "No taskcluster jobs.." to a PR once part2 and part3

### PR DESCRIPTION
**Pull request by iFlameing**
Bugzilla Bug: [1446768](https://bugzilla.mozilla.org/show_bug.cgi?id=1446768)

This pr deals with the second and third part. I didn't understand how I can implement the flag variable(which can disable this bug completely) in taskcluster.yml in the master branch.

@owlishDeveloper can you describe me, how we implemented the flag for allowPullRequests.

### CODE REVIEW

File: services/github/src/handlers.js [hunk 1]
```diff
@@ -485,6 +485,60 @@ async function jobHandler(message) {
   debug(`handling ${message.payload.details['event.type']} webhook for: ${organization}/${repository}@${sha}`);
   let repoconf = undefined;
 
+  // Checking pull request permission.
```
[the remark this answers is not in the document]
Comment (iFlameing) [reply services/github/src/handlers.js]: I amend a new commit with the changes you suggested.

**Comment by iFlameing** [comment]
@owlishDeveloper I changed my logic as you suggested for part2. can you review this once again?
And how I can solve part1 of this issue?

**Comment by owlishDeveloper** [comment]
> And how I can solve part1 of this issue?

I think that best done in a separate PR. Let's get this one landed. Nice work, thank you!

**Comment by iFlameing** [comment]
@owlishDeveloper  thanks

**Comment by owlishDeveloper** [comment]
> Why does this change where in the function we check for permission? I feel like we have tried to change the order of operations here before, and gotten in trouble.

You didn't mention this when we were discussing this a couple of days ago. The reason I want this to be where it is in the PR is that we are doing a whole bunch of work (reading and parsing YML in the PR's branch, generating the task tree, etc.) - why perform this work if PR is not allowed?

> Will this end up looking for a `.taskcluster.yml` when there isn't one, for example?

There is a `try-catch` block for that.

**Comment by owlishDeveloper** [comment]
> I only vaguely remember the trouble.

I see no such PR in git history. That logic has always been where it is now (unless you tried deploying from another branch and that PR was not merged. If there will be any trouble, we will return the logic back and leave a comment.

**Comment by edmorley** [comment]
Hi! 

The taskcluster bot has started commenting on our PRs in the last 24 hours, when we don't use taskcluster to run CI, so don't want any comments:
https://github.com/mozilla/treeherder/pull/4645#issuecomment-464151584

Is it possible this change is the cause?

**Comment by edmorley** [comment]
I've filed the above as:
https://bugzilla.mozilla.org/show_bug.cgi?id=1528778

**Pull request merged**